pkg/localstorev2/subscribe_push.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storer
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/localstorev2/internal/upload"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const subscribePushEventKey = "subscribe-push"
+
+func (db *DB) SubscribePush(ctx context.Context) (chunks chan swarm.Chunk, stop func()) {
```

Comment:
We should keep the signature of the function the same. Pusher currently maintains a skipList for chunks. This prevents us from repeating same chunks over and over again. So we need to check if the chunk returned from the iteration is in the skipList and if yes we should move on to next chunk.

Also, we will need a repeat-type function here. This is because if we fail to push a chunk we need to start over again. Maybe this can be avoided, but then this would need changes in Pusher. It would be better if we can handle it here.

Reply:
we decided to remove this repeat function to reduce the avoidable complexity. 

instead, on error, we feed chunk back into the pusher worker as such:
https://github.com/ethersphere/bee/pull/3856/files#diff-a01b070dce6187e6c1adf4a7aa78408ad5bbd8e16d3fe741edde8a190a128f03R163